lib/widgets/emoji_reaction.dart

```diff
@@ -360,3 +366,261 @@ class _TextEmoji extends StatelessWidget {
+  required BuildContext pageContext,
+  required Message message,
+}) {
+  final store = PerAccountStoreWidget.of(pageContext);
+  showModalBottomSheet<void>(
+    context: pageContext,
+    // Clip.hardEdge looks bad; Clip.antiAliasWithSaveLayer looks pixel-perfect
+    // on my iPhone 13 Pro but is marked as "much slower":
+    //   https://api.flutter.dev/flutter/dart-ui/Clip.html
+    clipBehavior: Clip.antiAlias,
+    useSafeArea: true,
+    isScrollControlled: true,
+    builder: (BuildContext context) {
+      return SafeArea(
+        child: Padding(
+          // By default, when software keyboard is opened, the ListView
+          // expands behind the software keyboard — resulting in some
+          // list entries being covered by the keyboard. Add explicit
+          // bottom padding the size of the keyboard, which fixes this.
+          padding: EdgeInsets.only(bottom: MediaQuery.viewInsetsOf(context).bottom),
+          // For _EmojiPickerItem, and RealmContentNetworkImage used in ImageEmojiWidget.
+          child: PerAccountStoreWidget(
+            accountId: store.accountId,
+            child: EmojiPicker(pageContext: pageContext, message: message))));
+    });
+}
+
+@visibleForTesting
+class EmojiPicker extends StatefulWidget {
+  const EmojiPicker({
+    super.key,
+    required this.pageContext,
+    required this.message,
+  });
+
+  final BuildContext pageContext;
+  final Message message;
+
+  @override
+  State<EmojiPicker> createState() => _EmojiPickerState();
+}
+
+class _EmojiPickerState extends State<EmojiPicker> with PerAccountStoreAwareStateMixin<EmojiPicker> {
+  late TextEditingController _controller;
+
+  EmojiAutocompleteView? _viewModel;
+  List<EmojiAutocompleteResult> _resultsToDisplay = const [];
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = TextEditingController()
+      ..addListener(_handleControllerUpdate);
+  }
+
+  void _initViewModel(EmojiAutocompleteQuery query) {
+    final store = PerAccountStoreWidget.of(context);
+    _viewModel = EmojiAutocompleteView.init(store: store, query: query)
+      ..addListener(_handleViewModelUpdate);
+  }
+
+  @override
+  void onNewStore() {
+    if (_viewModel != null) {
+      _viewModel!.dispose();
+    }
+    _initViewModel(EmojiAutocompleteQuery(_controller.text));
+  }
+
+  void _handleControllerUpdate() {
+    if (_viewModel != null) {
+      _viewModel!.query = EmojiAutocompleteQuery(_controller.text);
+    }
+  }
+
+  void _handleViewModelUpdate() {
+    setState(() {
+      _resultsToDisplay = List.unmodifiable(_viewModel!.results);
+    });
+  }
+
+  @override
+  void dispose() {
+    _viewModel?.dispose();
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final designVariables = DesignVariables.of(context);
+
+    return Column(
+      mainAxisSize: MainAxisSize.min,
```

Comment:
Does this do anything when there's an `Expanded` child? I'd think it doesn't.

Reply:
Yeah it doesn't, removed.